src/tests/e2etest/ZtsiTests.cs

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+using Microsoft.Azure.Devices.Shared;
+using Microsoft.Azure.Devices;
+using System.Text.Json;
+using System;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+
+namespace E2eTesting
+{
+    [TestFixture]
+    public class ZtsiTests : E2eTest
+    {   
+        const string ComponentName = "Ztsi";
+        public partial class Ztsi
+        {
+            public int Enabled { get; set; }
+            public string ServiceUrl { get; set; }
+        }
+        public partial class DesiredZtsi
+        {
+            public bool DesiredEnabled { get; set; }
+            public string DesiredServiceUrl { get; set; }
+        }
+
+        public partial class ResponseCode
+        {
+            public int ac { get; set; }
+        }
+        private int m_twinTimeoutSeconds;
+
+        [SetUp]
+        public void TestSetUp()
+        {
+            m_twinTimeoutSeconds = base.twinTimeoutSeconds;
+            base.twinTimeoutSeconds = base.twinTimeoutSeconds * 2;
```

Comment:
why does this test require a longer timeout?

Reply:
In practice, the extended time out seems to be needed for ztsi tests